app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/splash/SplashViewModel.kt

```diff
@@ -3,34 +3,74 @@ package com.kks.nimblesurveyjetpackcompose.viewmodel.splash
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.kks.nimblesurveyjetpackcompose.di.IoDispatcher
+import com.kks.nimblesurveyjetpackcompose.model.ResourceState
 import com.kks.nimblesurveyjetpackcompose.repo.login.LoginRepo
+import com.kks.nimblesurveyjetpackcompose.util.extensions.ErrorType
+import com.kks.nimblesurveyjetpackcompose.util.extensions.mapError
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 var SPLASH_TIME = 2000L
 
-@Suppress("UnusedPrivateMember")
 @HiltViewModel
 class SplashViewModel @Inject constructor(
     private val loginRepo: LoginRepo,
     @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
 ) : ViewModel() {
-
+    private val _isLoading = MutableStateFlow(false)
     private val _shouldNavigateToLogin = MutableStateFlow(false)
+    private val _isLoginSuccess = MutableStateFlow(false)
+    private var _isError = MutableStateFlow(ErrorType.NONE to "")
+
+    val isError: StateFlow<Pair<ErrorType, String>>
```

Comment:
We should rename them. `isError` is a name for a `Boolean` property by convention. We can have a UI model for error to keep the states. Also, what the `String` value represents is not clear. What do you think?